internal/manifests/collector/service.go

```diff
@@ -108,6 +108,34 @@ func MonitoringService(params manifests.Params) (*corev1.Service, error) {
 	}, nil
 }
 
+func ExtensionService(params manifests.Params) (*corev1.Service, error) {
+	name := naming.Service(params.OtelCol.Name)
+	labels := manifestutils.Labels(params.OtelCol.ObjectMeta, name, params.OtelCol.Spec.Image, ComponentOpenTelemetryCollector, []string{})
+
+	annotations, err := manifestutils.Annotations(params.OtelCol, params.Config.AnnotationsFilter())
+	if err != nil {
+		return nil, err
+	}
+
+	ports, err := params.OtelCol.Spec.Config.GetExtensionPorts(params.Log)
```

Comment:
If there are no ports... maybe we can just not create the service

Reply:
+1

Reply:
Do you mean something similar to this?
```
if len(ports) == 0 {
    return nil, nil
}
```

Reply:
Yes. That's what I meant.

I'm not sure about the HA. Do we need GRPC for this?

Reply:
> Yes. That's what I meant.

I have already implemented that and written proper unit tests for the same. 
Although, I have a suggestion/doubt. Won't it be a good idea to add labels in the extension service similar to [these](https://github.com/open-telemetry/opentelemetry-operator/blob/main/internal/manifests/collector/service.go#L32-L36)?

Reply:
> Do we need GRPC for this?

@yurishkuro can you answer this?